src/main/java/Problem1/ValidParentheses.java

```diff
@@ -1,11 +1,46 @@
 package Problem1;
 
-public class ValidParentheses {
+public class  ValidParentheses {
 
-    // Do not change signature (function name, parameters, return type)
-    public static boolean isValid(String str) {
-        // homework
-        // !!! must use ArrayStack or LinkedListStack from problem 1
-        return false; // place holder
+    public static boolean isValid(String str)
+    {
+        if(str == null){
+            return true;
+        }
+        ArrayStack<Character> stack = new ArrayStack(str.length());
+        for (int i = 0; i < str.length(); i++){
+            char c = str.charAt(i);
+            if( c == '(' || c == '{'|| c == '[')
+            {
+                stack.push(c);
+            }
+            else if( c == ')') {
+                if(stack.size() == 0 || stack.pop() != '(')
+                {
+                    return false;
+                }
+            }
+            else if( c == '}') {
+                if(stack.size() == 0 || stack.pop() != '{')
+                {
+                    return false;
+                }
+            }
+            else if( c == ']')
```

Comment:
the condition check here is quite complicated. this makes it hard to read and verify logic. try and see if this can refactored into less level of condition check.

Reply:
here's what I have for this part. just for reference and comparision.

```java 
   public static boolean isValid(String str) {
        if (str == null || str.isEmpty() || str.trim().length() == 0) {
            return true;
        }

        Stack<Character> stack = new LinkedListStack<>();

        for (int i = 0; i < str.length(); i++) {
            char c = str.charAt(i);
            Character left = stack.peek();
            if (left == null || !characterMatch(left, c)) {
                stack.push(c);
                continue;
            }
            stack.pop();
        }
        return stack.size() == 0;
    }

    private static boolean characterMatch(char left, char right) {
        switch (left) {
            case '(':
                return right == ')';
            case '{':
                return right == '}';
            case '[':
                return right == ']';
        }
        return false;
    }
```